docfiles/docs.js

```diff
@@ -259,6 +260,72 @@ function renderSnippets() {
     });
 }
 
+function getCookieLang() {    
+    const cookiePropRegex = new RegExp(`${pxt.Util.escapeForRegex(pxt.Util.pxtLangCookieId)}=(.*?)(?:;|$)`)
+    const cookieValue = cookiePropRegex.exec(document.cookie);
+    return cookieValue && cookieValue[1] || null;
+}
+
+function setCookieLang(langId) {
+    if (!pxt.Util.allLanguages[langId]) {
+        return;
+    }
+    if (langId !== getCookieLang()) {
+        pxt.tickEvent(`menu.lang.setcookielang.${langId}`);
+        const expiration = new Date();
+        expiration.setTime(expiration.getTime() + (pxt.Util.langCookieExpirationDays * 24 * 60 * 60 * 1000));
+        document.cookie = `${pxt.Util.pxtLangCookieId}=${langId}; expires=${expiration.toUTCString()}`;
+    }
+}
+
+function languageOption(code) {
+    let locale = pxt.Util.allLanguages[code];
+    return `<div class="ui card link card-selected langoption" data-lang="${code}" role="link" aria-label="${locale.englishName}" 
+        tabindex="0"><div class="content"><div class="header">${locale.localizedName}</div>
+        <div class="description tall">${locale.englishName}</div></div></div>`
+}
+
+function setupLangPicker() {
+
+    let availableLocales = pxt.appTarget.appTheme.availableLocales;
+    let locales;
+    let initialLang = pxt.Util.normalizeLanguageCode(getCookieLang())[0];
+    if (pxt.appTarget.appTheme && availableLocales) {
+
+        locales = availableLocales.map(e => languageOption(e)).join("\n");
+        document.querySelector("#availablelocales").innerHTML = locales;
```

Comment:
Don't use innerHTML, if you need to inflate HTML we usually use `document.createElement()` directly and `el.textcontent = "whatever"`

Reply:
good practice, thanks